### DISCUSSION TITLE
White margin edit in printout

### DISCUSSION THREAD

**Discussion by petergie** [sumatrapdfreader/sumatrapdf#3566] [Q&A]
Hi!
I want to print a file with miy canon MX925 in brochure mode. In the GUI it is ok, but in the printout there is a broad white margin added, and the text is nunnecessary small. The filee is ok, and printing with another viewer is ok. I can´t find any adjustment in Sumatra to change this.

Thx
peter
![Druck1](https://github.com/sumatrapdfreader/sumatrapdf/assets/13985554/a9369a54-56d4-49ad-8cfd-5cfb6b757c1a)
![Druck2](https://github.com/sumatrapdfreader/sumatrapdf/assets/13985554/51e6f1f8-4ff1-4056-bc35-5fb964ed5a8f)

**Comment by GitHubRulesOK** [comment]
SumatraPDf has limited commands to control Printing as its down to the driver
The 3 options are none shrink (usually helps with scale down) fit (usually helps with scale up) in all cases margin controls are in the printing driver, which may not be as expected depending on media settings at both ends.

1st issue is auto rate may be needed and that will be printing images centred default top left corner by virtue of that is the datum point for scaling.

A3 Landscape printed as A4 portrait
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/a5097b97-4613-49da-a138-444b5b857010)
A4 portrait printed to A3 Landscape
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/8006c5bc-6529-47b3-a082-cdda348b0812)

**Reply by petergie** [reply]
These options are only active  in the GUI, not in printing. And the problem seems not to be in the driver, because printing with the same options out of PDF XChange Editor works well.

**Reply by GitHubRulesOK** [reply]
That document is smaller that standard a4 thus for booklet printing the driver needs to be set to n-up 2 pages AND split as 2 x FIT so on command line printing `fit` needs to be used for upscaling the source

**Reply by petergie** [reply]
Yes, the document is exactly  one half of A4, but in booklet printing mode the driver prints two pages on one A4 paper. Thre is no scaling necessesary. I still don´t understand, why only sumatra makes this useless margins ☹

**Reply by GitHubRulesOK** [reply]
its not a margin (pdf doesnot have any they are nothings around the body for fitting to A5 paper kind) it is a boundary that needs to fit the printing boundary thus it needs upscale by fit when reducing A5 downwards on A4 media. Note SumatraPDF does not scale down it is the print driver that is doing the reduction all SumatraPDF can do is allow reduce or hint FIT TO PRINT AREA.

The driver has "borderless" or "margin" settings that must also be 0 to get the maximum scale. The opposite can also happen where the driver crops the void areas which is just as confusing as the ink bleeds out of the print zone. But in either direction it is not SumatraPDF making any adjustment it does not edit the print boundary!

printout of ONE FIT page to 2 PAGES per page at draft quality so cannot be borderless and printer decides gripper marginals (here will be at top and bottom).
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/199145ac-6b39-431a-a7b4-2a1379ba1fc8)

**Comment by GitHubRulesOK** [comment]
Most PDF editors include a printing engine So Acrobat is the best especially with Distiller PS/PDF rewriter then GhostScript is perfect for Command line PDL Printing just difficult to use PS to trim contents then other editors such as Trackers products (Which includes its own print driver) Then perhaps Foxit reader/writer and then other viewers with vector text support like say a web browser (Adobe Acrobat Edge or Chromium Foxit/PDFium engines) 

SumatraPDF depends entirely on MS managing print defaults and it does that by handing limited changes over to the driver as Graphics not Page Description Language, images usually are lossy and still big by comparison to a graphics editor app.
